README.md

```diff
@@ -0,0 +1,321 @@
+
+This class is not required in any Captain waypoint. If you do not need to track the request id of your work, you don't need to implement this class. If this is the case, you can just have your Request Submitter return null as it won't be read anywhere else.
+
+###### Status Retriever
+
+Implementing `CaptainStatusRetriever` allow you to tell a Captain Waypoint how to check the status of a request. It takes in an id and returns a `CaptainStatus`.
+
+The most common use case is to poll a service to which you've submitted work as to the status of your request. Based on the status returned by the service, you tell Captain what status it should set for your request.
+
+##### Waypoint Types
+
+There are 3 types of waypoints. Each use some or all of these Waypoint Components described above (Request Submitter, Handle Persistor, Status Retriever)
+
+###### Asynchronous Waypoint
+
+This Waypoint submits work to a service and saves the request id it receives from the service. It then uses the status retriever to check back that that service fulfilled its request. It accepts a Request Submitter, Handle Persistor, and Status Retriver. The Handle Persistor is optional and can be skipped in the cases described in the Handle Persistor section.
+
+e.g. I submit a request to the analytics service that kicks off some long running job. Save the job id. I poll until my job is done.
+
+note: "submits work to a service" technically your waypoint can just do work in the process it's spun up on whatever local hardware its running on. Doing so subverts the point of Captain. The goal is to make coordination between components cheap so that we make good choices in separating our concerns. As a rule of thumb, the less code you're writing in the Captain classes, the more you're adhering to the intended goals of the project. Nothing is stopping you from doing horrible things, Captain's trying to give you every opportunity to make a good choice.
+
+###### Synchronous Waypoint
+
+This Wayoint submits work to a service, and then, as long as the submission of that work does not fail, moves the request on to the next step in your manifest.
+
+It accepts a Request Submitter. 
+
+It optionally accepts a Handle Persistor, in the case where you want to save a request id, but don't want to wait for that request to be processed. 
+
+e.g. I submit a request to a service to report that new stats have been generated. As long as the service returns no error, I'm good to go.
+
+###### Control Flow Waypoint
+ 
+This Waypoint is designed for:  
+1. forcing a request to wait for pre-conditions to be met before progressing in your pipeline. 
+2. making it easy to add validations to your Captain workflow.
+
+It accepts a Status Retriever only. Here are a couple examples of how it could be used:
+
+e.g. Hold the request until the PO has completed configuration of the CustomerLinkAccount.
```

Comment:
Not sure if we want to use different wording instead of PO :P might not be obvious to everyone else. Same with `CustomerLinkAccount`. Doesn't seem to be seen in the code so doesn't really align with whats been written.

Reply:
bah. typo. thanks.